android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2437,16 +2436,32 @@ public static void applyKeyboardHeight(Context context, int height, int orientat
       KeyboardHeight_Context_Portrait_Current = height;
     }
     editor.commit();
+
+    // Track whether height was successfully applied
+    boolean heightApplied = false;
+
     // Confirm new LayoutParams for in-app or system keyboards
     if (isKeyboardLoaded(KeyboardType.KEYBOARD_TYPE_INAPP)) {
       InAppKeyboard.loadJavascript(KMString.format("setOskHeight('%s')", height));
       RelativeLayout.LayoutParams params = getKeyboardLayoutParams();
       InAppKeyboard.setLayoutParams(params);
+      heightApplied = true;
     }
     if (isKeyboardLoaded(KeyboardType.KEYBOARD_TYPE_SYSTEM)) {
       SystemKeyboard.loadJavascript(KMString.format("setOskHeight('%s')", height));
       RelativeLayout.LayoutParams params = getKeyboardLayoutParams();
       SystemKeyboard.setLayoutParams(params);
+      heightApplied = true;
+    }
+
+    // If keyboard wasn't loaded, set a pending flag so height gets applied when keyboard loads
+    if (!heightApplied) {
```

Comment:
I am not sure if this is 100% correct. What if only the in-app keyboard is loaded: then its layout params will be updated, but the system keyboard layout params won't be, and we won't ever receive the flag to update its layout params. And vice-versa. Do we need to track a pending flag separately for in-app vs system?

Reply:
This was already an intermittent bug to reproduce (it shows up most often when using android Fragments that get disposed on aggressive rotation without triggering KB webview refreshes). The Keyman app uses Activities that are less fragile. After the making this fix to the local copy of Keyman 18 in app builders, I couldn't reproduce the bug. I'll defer to you on this question.